stellargraph/data/explorer.py

```diff
@@ -451,24 +435,26 @@ def transition_probability(nn):
                 current_node = node
 
                 for _ in range(length - 1):
-                    neighbours = self.graph.neighbors(
-                        current_node, use_ilocs=True, include_edge_weight=weighted
+                    # select one of the neighbours using the
+                    # appropriate transition probabilities
+                    neighbours, weights = self.graph.neighbor_arrays(
+                        current_node, include_edge_weight=True, use_ilocs=True
                     )
 
-                    if not neighbours:
+                    if len(neighbours) == 0:
                         break
 
-                    # select one of the neighbours using the
-                    # appropriate transition probabilities
-                    choice = naive_weighted_choices(
-                        rs, (transition_probability(nn) for nn in neighbours),
-                    )
+                    mask = neighbours == previous_node
+                    weights[mask] *= ip
```

Comment:
We're sure that this will never be mutating the weights inside the `EdgeData`, right?

Reply:
yep, looking up an array with a list/array indices always returns a copy even if the indices are sequeiential e.g.

```
import numpy as np

arr = np.arange(5)
copy = arr[[0, 1, 2, 4]]
copy[:] = 0
assert (copy == 0).all()
assert not (arr == 0).all()
```

will work